components/conversations/EmojiReactionPicker.js

```diff
@@ -58,34 +72,43 @@ const ReactionButton = styled(StyledRoundButton).attrs({ isBorderless: true, but
     `}
 `;
 
-const getOptimisticResponse = (comment, emoji, isAdding) => {
-  const userReactions = comment.userReactions || [];
+const getOptimisticResponse = (entity, emoji, isAdding) => {
+  const userReactions = entity.userReactions || [];
+  const { __typename } = entity;
+  const fieldName = __typename === 'Update' ? 'update' : 'comment';
   if (isAdding) {
-    const newCount = (comment.reactions[emoji] || 0) + 1;
+    const newCount = (entity.reactions[emoji] || 0) + 1;
+
     return {
       __typename: 'Mutation',
-      addCommentReaction: {
-        __typename: 'Comment',
-        id: comment.id,
-        reactions: { ...comment.reactions, [emoji]: newCount },
-        userReactions: [...userReactions, emoji],
+      addEmojiReaction: {
+        __typename: 'addEmojiReaction',
```

Comment:
To be confirmed by looking at the actual mutation response in the Network logs, but I think we want:

```suggestion
        __typename: 'EmojiReactionsResponse',
```

Reply:
The network logs show, 

```
{"data":{"addEmojiReaction":{"update":{"id":"medlzx7y-va6ep4jn-dbr89wg0-omnq3kqo","reactions":{"🎉":1,"👎":1,"😀":1,"❤️":1,"👍️":1},"userReactions":["👍️","❤️","😀","👎","🎉"],"__typename":"Update"},"comment":null,"__typename":"EmojiReactionOutput"}}}
```

So I think it should be `EmojiReactionOutput`. 🤔 Is my understanding correct? 😄

Reply:
Renamed this as well as in API side. 😄